acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+    print("============================================")
+    with open(te_stitch_file) as f:
+        lines = f.readlines()
+
+    # Calculate number of storms and max length
+    num_storms, max_len = _calc_num_storms_and_max_len(lines)
+    # Parse variables from TE stitch file
+    te_stitch_vars = _get_vars_from_te_stitch(lines, max_len, num_storms)
+
+    # Use E3SM land-sea mask
+    mask_path = os.path.join(acme_diags.INSTALL_PATH, "acme_ne30_ocean_land_mask.nc")
+    ocnfrac = cdms2.open(mask_path)("OCNFRAC", squeeze=1)
+
+    # From model data, this dict stores a tuple for each basin.
+    # (mean ace, tc_intensity_dist, seasonal_cycle, # storms, # of storms over the ocean)
+    result_mod: Dict[str, Any] = {}
+    result_mod["num_years"] = te_stitch_vars["num_years"]
+
+    for basin, basin_info in BASIN_DICT.items():
+        mod_vars = _get_mod_vars(
+            num_storms,
+            te_stitch_vars,
+            ocnfrac,
+            basin_info,
+        )
+
+        # Calculate storm intensity and seasonal cycle
+        pdf_mod_intensity = _calc_ts_intensity_dist(mod_vars["mod_wnd"])
+        pdf_mod_seasonal_cycle = _calc_seasonal_cycle(mod_vars["mod_mon"])
+
+        # Calculate storms and storms over ocean
+        storms = mod_vars["mod_num"] / te_stitch_vars["num_years"]
+        storms_over_ocean = mod_vars["mod_num_ocn"]
+
+        result_mod[basin] = [
+            mod_vars["mod_ace_mean"],
+            pdf_mod_intensity,
+            pdf_mod_seasonal_cycle,
+            storms,
+            storms_over_ocean,
+        ]
+
+    return result_mod
+
+
+def _calc_num_storms_and_max_len(lines: List[str]) -> Tuple[int, int]:
+    """Calculate number of storms and max length using lines from a TE stitch file.
+
+    :param lines: Lines from TE stitch file
+    :type lines: List[str]
+    :return: Number of storms and max storm length
+    :rtype: Tuple[int, int]
+    """
+    num_storms = 0
+    max_len = 0
+
+    for i in range(0, np.size(lines)):
+        if lines[i][0] == "s":
+            num_storms = num_storms + 1
+            num_points = 0
+        else:
+            num_points = num_points + 1
+            max_len = max(max_len, num_points)
+
+    print("Number of storms:", num_storms)
+    print("Max length of storms:", max_len)
+    return num_storms, max_len
+
+
+def _get_vars_from_te_stitch(
+    lines: List[str], max_len: int, num_storms: int
+) -> Dict[str, Any]:
+    """Extracts variables from lines of a TE stitch file.
+
+    :param lines: Lines from a TE stitch file
+    :type lines: List[str]
+    :param max_len: Max length of storms
+    :type max_len: int
+    :param num_storms: Number of storms
+    :type num_storms: int
+    :return: Dictionary of variables from TE stitch file
+    :rtype: Dict[str, Any]
+    """
+    keys = ("longmc", "latmc", "vsmc", "yearmc", "monthmc")
+    vars_dict = {k: np.empty((max_len, num_storms)) * np.nan for k in keys}
```

Comment:
These were individual variables with the same initial value `np.empty((max_len, num_storms)) * np.nan`.
Making a dictionary reduces this redundancy.